src/views/ListOfBooks.jsx

```diff
@@ -33,18 +35,36 @@ const ListOfBooks = () => {
     });
   },[]);
 
+  // Filter books by subject
+  const subjects = books?.flatMap((book) => book.subjects?.map((subject) => subject.name));
+
+  const onlyUnique = (value, index, self) => {
+    return self.indexOf(value) === index;
+  }
+
+  // Get unique subjects to display in the select option
+  const subjectsList = subjects?.filter(onlyUnique);
```

Comment:
This is a neat trick, but the second recipe (using a `Set`) in the SO post is more efficient, both from the point of view of conciseness and performance https://stackoverflow.com/a/14438954/304876.